src/app/content/components/Sidebar.tsx

```diff
@@ -194,43 +327,68 @@ export class Sidebar extends Component<SidebarProps> {
 
   public componentDidUpdate() {
     this.scrollToSelectedPage();
+    this.expandCurrentChapter();
   }
 
   private scrollToSelectedPage() {
     scrollTocSectionIntoView(this.sidebar, this.activeSection);
   }
 
-  private renderTocNode = (book: Book, {contents}: ArchiveTree) => <nav>
-    <ol>
-      {contents.map((item) => {
-        const active = (!!this.props.page) && stripIdVersion(item.id) === this.props.page.id;
-
-        return isArchiveTree(item)
-          ? <NavItem key={item.id}>
-              <h3 dangerouslySetInnerHTML={{__html: item.title}} />
-              {this.renderTocNode(book, item)}
-            </NavItem>
-          : <NavItem
-              key={item.id}
-              ref={active ? ((ref: any) => this.activeSection = ref) : undefined}
-              active={active}
-            >
-              <ContentLink
-                onClick={this.props.onNavigate}
-                book={book}
-                page={item}
-                dangerouslySetInnerHTML={{__html: item.title}}
-              />
-            </NavItem>;
-      })}
-    </ol>
-  </nav>
+  private expandCurrentChapter() {
+
+    if (typeof(document) === 'undefined' || typeof(window) === 'undefined') {
+      return;
+    }
+
+    const currentChapter: HTMLElement = document.querySelector('[aria-label="Current Page"]') as HTMLElement;
+
+    if ( currentChapter
```

Comment:
since units are a thing, what you actually want to do here is crawl up the dom until you reach `container` and assign `open` to any `details` elements that you encounter along the way. you can do this recursively and it'll be way less brittle than what is here now